### ISSUE TITLE
Validate bundles as they come in

### ISSUE THREAD

**Issue by GalRogozinski** [iotaledger/iri#1810]
### Description
We want to validate bundles as they come in and not only on tip-selection and milestone solidification

### Motivation
As part of the white-flag changes we are doing now, we expect very little of the tangle to be left-behind. So it makes sense to validate as they arrive. We can use the validation status to give the proper score to a tip in #1771

### Requirements
1. When a tail solidifies, and all the bundles it approves are valid, pass it on to bundle validation
2. We  have 3 possible validity statuses on a tail: INVALID, VALID, and UNKNOWN (we currently have this). Once a tail changes its status from UNKNOWN->VALID we start propagating the check upwards to the approvers. If it changes its status from UNKNOWN->INVALID, propagate the status to the approvers.

**Comment by GalRogozinski** [comment]
We should add a bundle validation stage after the solidification stage (if it exists)
or if not ready after received stage.
Currently this stage will not be a prerequisite to any other stages.
In Later issues we may change this so we only broadcast valid bundles (but not now)

**Comment by acha-bill** [comment]
This is blocked by #1805 and #1821
